// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/DataProjectionQuery.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataProjectionQuery = void 0;
/**
 * Query configuration for a data projection request.
 */
class DataProjectionQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DataProjectionQuery.attributeTypeMap;
    }
}
exports.DataProjectionQuery = DataProjectionQuery;
/**
 * @ignore
 */
DataProjectionQuery.attributeTypeMap = {
    dataSource: {
        baseName: "data_source",
        type: "string",
        required: true,
    },
    indexes: {
        baseName: "indexes",
        type: "Array<string>",
    },
    queryString: {
        baseName: "query_string",
        type: "string",
        required: true,
    },
    storage: {
        baseName: "storage",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
