// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/PointPlotWidgetRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PointPlotWidgetRequest = void 0;
/**
 * Request configuration for the point plot widget.
 */
class PointPlotWidgetRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PointPlotWidgetRequest.attributeTypeMap;
    }
}
exports.PointPlotWidgetRequest = PointPlotWidgetRequest;
/**
 * @ignore
 */
PointPlotWidgetRequest.attributeTypeMap = {
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    projection: {
        baseName: "projection",
        type: "PointPlotProjection",
        required: true,
    },
    query: {
        baseName: "query",
        type: "DataProjectionQuery",
        required: true,
    },
    requestType: {
        baseName: "request_type",
        type: "DataProjectionRequestType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
