// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/PointPlotWidgetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PointPlotWidgetDefinition = void 0;
/**
 * The point plot displays individual data points over time.
 */
class PointPlotWidgetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PointPlotWidgetDefinition.attributeTypeMap;
    }
}
exports.PointPlotWidgetDefinition = PointPlotWidgetDefinition;
/**
 * @ignore
 */
PointPlotWidgetDefinition.attributeTypeMap = {
    customLinks: {
        baseName: "custom_links",
        type: "Array<WidgetCustomLink>",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    legend: {
        baseName: "legend",
        type: "PointPlotWidgetLegend",
    },
    markers: {
        baseName: "markers",
        type: "Array<WidgetMarker>",
    },
    requests: {
        baseName: "requests",
        type: "Array<PointPlotWidgetRequest>",
        required: true,
    },
    time: {
        baseName: "time",
        type: "WidgetTime",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    titleAlign: {
        baseName: "title_align",
        type: "WidgetTextAlign",
    },
    titleSize: {
        baseName: "title_size",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "PointPlotWidgetDefinitionType",
        required: true,
    },
    yaxis: {
        baseName: "yaxis",
        type: "WidgetAxis",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
