// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AppVersionAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppVersionAttributes = void 0;
/**
 * Attributes describing an app version.
 */
class AppVersionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AppVersionAttributes.attributeTypeMap;
    }
}
exports.AppVersionAttributes = AppVersionAttributes;
/**
 * @ignore
 */
AppVersionAttributes.attributeTypeMap = {
    appId: {
        baseName: "app_id",
        type: "string",
        format: "uuid",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    hasEverBeenPublished: {
        baseName: "has_ever_been_published",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        format: "date-time",
    },
    userId: {
        baseName: "user_id",
        type: "number",
        format: "int64",
    },
    userName: {
        baseName: "user_name",
        type: "string",
    },
    userUuid: {
        baseName: "user_uuid",
        type: "string",
        format: "uuid",
    },
    version: {
        baseName: "version",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
