// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ArbitraryRuleStatusResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArbitraryRuleStatusResponseData = void 0;
/**
 * Processing status for a custom allocation rule.
 */
class ArbitraryRuleStatusResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ArbitraryRuleStatusResponseData.attributeTypeMap;
    }
}
exports.ArbitraryRuleStatusResponseData = ArbitraryRuleStatusResponseData;
/**
 * @ignore
 */
ArbitraryRuleStatusResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ArbitraryRuleStatusResponseDataAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ArbitraryRuleStatusResponseDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
