// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudInventorySyncConfigAWSRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudInventorySyncConfigAWSRequestAttributes = void 0;
/**
 * AWS settings for the S3 bucket Storage Management reads inventory reports from.
 */
class CloudInventorySyncConfigAWSRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudInventorySyncConfigAWSRequestAttributes.attributeTypeMap;
    }
}
exports.CloudInventorySyncConfigAWSRequestAttributes = CloudInventorySyncConfigAWSRequestAttributes;
/**
 * @ignore
 */
CloudInventorySyncConfigAWSRequestAttributes.attributeTypeMap = {
    awsAccountId: {
        baseName: "aws_account_id",
        type: "string",
        required: true,
    },
    destinationBucketName: {
        baseName: "destination_bucket_name",
        type: "string",
        required: true,
    },
    destinationBucketRegion: {
        baseName: "destination_bucket_region",
        type: "string",
        required: true,
    },
    destinationPrefix: {
        baseName: "destination_prefix",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
