// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudInventorySyncConfigResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudInventorySyncConfigResponseData = void 0;
/**
 * Storage Management configuration data.
 */
class CloudInventorySyncConfigResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudInventorySyncConfigResponseData.attributeTypeMap;
    }
}
exports.CloudInventorySyncConfigResponseData = CloudInventorySyncConfigResponseData;
/**
 * @ignore
 */
CloudInventorySyncConfigResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CloudInventorySyncConfigAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "CloudInventorySyncConfigResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
