// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudInventorySyncConfigAzureRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudInventorySyncConfigAzureRequestAttributes = void 0;
/**
 * Azure settings for the storage account and container with inventory data.
 */
class CloudInventorySyncConfigAzureRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudInventorySyncConfigAzureRequestAttributes.attributeTypeMap;
    }
}
exports.CloudInventorySyncConfigAzureRequestAttributes = CloudInventorySyncConfigAzureRequestAttributes;
/**
 * @ignore
 */
CloudInventorySyncConfigAzureRequestAttributes.attributeTypeMap = {
    clientId: {
        baseName: "client_id",
        type: "string",
        required: true,
    },
    container: {
        baseName: "container",
        type: "string",
        required: true,
    },
    resourceGroup: {
        baseName: "resource_group",
        type: "string",
        required: true,
    },
    storageAccount: {
        baseName: "storage_account",
        type: "string",
        required: true,
    },
    subscriptionId: {
        baseName: "subscription_id",
        type: "string",
        required: true,
    },
    tenantId: {
        baseName: "tenant_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
