// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudInventorySyncConfigAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudInventorySyncConfigAttributes = void 0;
/**
 * Attributes for a Storage Management configuration. Fields other than `id` may be empty in the response immediately after a create or update; subsequent reads return the full configuration.
 */
class CloudInventorySyncConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudInventorySyncConfigAttributes.attributeTypeMap;
    }
}
exports.CloudInventorySyncConfigAttributes = CloudInventorySyncConfigAttributes;
/**
 * @ignore
 */
CloudInventorySyncConfigAttributes.attributeTypeMap = {
    awsAccountId: {
        baseName: "aws_account_id",
        type: "string",
        required: true,
    },
    awsBucketName: {
        baseName: "aws_bucket_name",
        type: "string",
        required: true,
    },
    awsRegion: {
        baseName: "aws_region",
        type: "string",
        required: true,
    },
    azureClientId: {
        baseName: "azure_client_id",
        type: "string",
        required: true,
    },
    azureContainerName: {
        baseName: "azure_container_name",
        type: "string",
        required: true,
    },
    azureStorageAccountName: {
        baseName: "azure_storage_account_name",
        type: "string",
        required: true,
    },
    azureTenantId: {
        baseName: "azure_tenant_id",
        type: "string",
        required: true,
    },
    cloudProvider: {
        baseName: "cloud_provider",
        type: "CloudInventoryCloudProviderId",
        required: true,
    },
    error: {
        baseName: "error",
        type: "string",
        required: true,
    },
    errorCode: {
        baseName: "error_code",
        type: "string",
        required: true,
    },
    gcpBucketName: {
        baseName: "gcp_bucket_name",
        type: "string",
        required: true,
    },
    gcpProjectId: {
        baseName: "gcp_project_id",
        type: "string",
        required: true,
    },
    gcpServiceAccountEmail: {
        baseName: "gcp_service_account_email",
        type: "string",
        required: true,
    },
    prefix: {
        baseName: "prefix",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
