// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CloudInventorySyncConfigGCPRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudInventorySyncConfigGCPRequestAttributes = void 0;
/**
 * GCP settings for buckets involved in inventory reporting.
 */
class CloudInventorySyncConfigGCPRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CloudInventorySyncConfigGCPRequestAttributes.attributeTypeMap;
    }
}
exports.CloudInventorySyncConfigGCPRequestAttributes = CloudInventorySyncConfigGCPRequestAttributes;
/**
 * @ignore
 */
CloudInventorySyncConfigGCPRequestAttributes.attributeTypeMap = {
    destinationBucketName: {
        baseName: "destination_bucket_name",
        type: "string",
        required: true,
    },
    projectId: {
        baseName: "project_id",
        type: "string",
        required: true,
    },
    serviceAccountEmail: {
        baseName: "service_account_email",
        type: "string",
        required: true,
    },
    sourceBucketName: {
        baseName: "source_bucket_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
