// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsUtilizationTimeseriesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsUtilizationTimeseriesResponse = void 0;
/**
 * Response containing timeseries utilization metrics for cloud commitment programs.
 */
class CommitmentsUtilizationTimeseriesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsUtilizationTimeseriesResponse.attributeTypeMap;
    }
}
exports.CommitmentsUtilizationTimeseriesResponse = CommitmentsUtilizationTimeseriesResponse;
/**
 * @ignore
 */
CommitmentsUtilizationTimeseriesResponse.attributeTypeMap = {
    series: {
        baseName: "series",
        type: "{ [key: string]: Array<number>; }",
        required: true,
    },
    times: {
        baseName: "times",
        type: "Array<number>",
        required: true,
    },
    unit: {
        baseName: "unit",
        type: "CommitmentsUnit",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
