// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsAwsSPCommitment.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsAwsSPCommitment = void 0;
/**
 * AWS Savings Plan commitment details.
 */
class CommitmentsAwsSPCommitment {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsAwsSPCommitment.attributeTypeMap;
    }
}
exports.CommitmentsAwsSPCommitment = CommitmentsAwsSPCommitment;
/**
 * @ignore
 */
CommitmentsAwsSPCommitment.attributeTypeMap = {
    commitmentId: {
        baseName: "commitment_id",
        type: "string",
        required: true,
    },
    committedSpendPerHour: {
        baseName: "committed_spend_per_hour",
        type: "number",
        format: "double",
    },
    expirationDate: {
        baseName: "expiration_date",
        type: "string",
    },
    purchaseOption: {
        baseName: "purchase_option",
        type: "string",
        required: true,
    },
    savingsPlanType: {
        baseName: "savings_plan_type",
        type: "string",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "string",
    },
    termLength: {
        baseName: "term_length",
        type: "number",
        format: "double",
    },
    utilization: {
        baseName: "utilization",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
