// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsUnit.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsUnit = void 0;
/**
 * Unit metadata for a numeric metric.
 */
class CommitmentsUnit {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsUnit.attributeTypeMap;
    }
}
exports.CommitmentsUnit = CommitmentsUnit;
/**
 * @ignore
 */
CommitmentsUnit.attributeTypeMap = {
    family: {
        baseName: "family",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "number",
        required: true,
        format: "int64",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    plural: {
        baseName: "plural",
        type: "string",
        required: true,
    },
    scaleFactor: {
        baseName: "scale_factor",
        type: "number",
        required: true,
        format: "double",
    },
    shortName: {
        baseName: "short_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
