// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsAwsEC2RICommitment.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsAwsEC2RICommitment = void 0;
/**
 * AWS EC2 Reserved Instance commitment details.
 */
class CommitmentsAwsEC2RICommitment {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsAwsEC2RICommitment.attributeTypeMap;
    }
}
exports.CommitmentsAwsEC2RICommitment = CommitmentsAwsEC2RICommitment;
/**
 * @ignore
 */
CommitmentsAwsEC2RICommitment.attributeTypeMap = {
    availabilityZone: {
        baseName: "availability_zone",
        type: "string",
    },
    commitmentId: {
        baseName: "commitment_id",
        type: "string",
        required: true,
    },
    expirationDate: {
        baseName: "expiration_date",
        type: "string",
    },
    instanceType: {
        baseName: "instance_type",
        type: "string",
        required: true,
    },
    numberOfNfus: {
        baseName: "number_of_nfus",
        type: "number",
        format: "double",
    },
    numberOfReservations: {
        baseName: "number_of_reservations",
        type: "number",
        format: "double",
    },
    offeringClass: {
        baseName: "offering_class",
        type: "string",
        required: true,
    },
    operatingSystem: {
        baseName: "operating_system",
        type: "string",
        required: true,
    },
    purchaseOption: {
        baseName: "purchase_option",
        type: "string",
        required: true,
    },
    region: {
        baseName: "region",
        type: "string",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "string",
    },
    termLength: {
        baseName: "term_length",
        type: "number",
        format: "double",
    },
    utilization: {
        baseName: "utilization",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
