// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsScalarColumn.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsScalarColumn = void 0;
/**
 * A column in a scalar response. When type is "group", values contains arrays of strings. When type is "number", values contains numeric values.
 */
class CommitmentsScalarColumn {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsScalarColumn.attributeTypeMap;
    }
}
exports.CommitmentsScalarColumn = CommitmentsScalarColumn;
/**
 * @ignore
 */
CommitmentsScalarColumn.attributeTypeMap = {
    meta: {
        baseName: "meta",
        type: "CommitmentsScalarColumnMeta",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "CommitmentsScalarColumnType",
        required: true,
    },
    values: {
        baseName: "values",
        type: "Array<any>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
