// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsAwsRDSRICommitment.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsAwsRDSRICommitment = void 0;
/**
 * AWS RDS Reserved Instance commitment details.
 */
class CommitmentsAwsRDSRICommitment {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsAwsRDSRICommitment.attributeTypeMap;
    }
}
exports.CommitmentsAwsRDSRICommitment = CommitmentsAwsRDSRICommitment;
/**
 * @ignore
 */
CommitmentsAwsRDSRICommitment.attributeTypeMap = {
    commitmentId: {
        baseName: "commitment_id",
        type: "string",
        required: true,
    },
    databaseEngine: {
        baseName: "database_engine",
        type: "string",
        required: true,
    },
    expirationDate: {
        baseName: "expiration_date",
        type: "string",
    },
    instanceType: {
        baseName: "instance_type",
        type: "string",
        required: true,
    },
    isMultiAz: {
        baseName: "is_multi_az",
        type: "boolean",
    },
    numberOfNfus: {
        baseName: "number_of_nfus",
        type: "number",
        format: "double",
    },
    numberOfReservations: {
        baseName: "number_of_reservations",
        type: "number",
        format: "double",
    },
    purchaseOption: {
        baseName: "purchase_option",
        type: "string",
        required: true,
    },
    region: {
        baseName: "region",
        type: "string",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "string",
    },
    termLength: {
        baseName: "term_length",
        type: "number",
        format: "double",
    },
    utilization: {
        baseName: "utilization",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
