// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsAzureVMRICommitment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsAzureVMRICommitment = void 0;
/**
 * Azure Virtual Machine Reserved Instance commitment details.
 */
class CommitmentsAzureVMRICommitment {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsAzureVMRICommitment.attributeTypeMap;
    }
}
exports.CommitmentsAzureVMRICommitment = CommitmentsAzureVMRICommitment;
/**
 * @ignore
 */
CommitmentsAzureVMRICommitment.attributeTypeMap = {
    benefitName: {
        baseName: "benefit_name",
        type: "string",
        required: true,
    },
    commitmentId: {
        baseName: "commitment_id",
        type: "string",
        required: true,
    },
    expirationDate: {
        baseName: "expiration_date",
        type: "string",
    },
    instanceType: {
        baseName: "instance_type",
        type: "string",
        required: true,
    },
    meterSubCategory: {
        baseName: "meter_sub_category",
        type: "string",
        required: true,
    },
    region: {
        baseName: "region",
        type: "string",
        required: true,
    },
    startDate: {
        baseName: "start_date",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "CommitmentsAzureVMRIStatus",
        required: true,
    },
    termLength: {
        baseName: "term_length",
        type: "number",
        format: "double",
    },
    utilization: {
        baseName: "utilization",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
