// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsSavingsTimeseriesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsSavingsTimeseriesResponse = void 0;
/**
 * Response containing timeseries savings metrics for cloud commitment programs.
 */
class CommitmentsSavingsTimeseriesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsSavingsTimeseriesResponse.attributeTypeMap;
    }
}
exports.CommitmentsSavingsTimeseriesResponse = CommitmentsSavingsTimeseriesResponse;
/**
 * @ignore
 */
CommitmentsSavingsTimeseriesResponse.attributeTypeMap = {
    actualCost: {
        baseName: "actual_cost",
        type: "CommitmentsTimeseriesMetric",
        required: true,
    },
    effectiveSavingsRate: {
        baseName: "effective_savings_rate",
        type: "CommitmentsTimeseriesMetric",
        required: true,
    },
    onDemandEquivalentCost: {
        baseName: "on_demand_equivalent_cost",
        type: "CommitmentsTimeseriesMetric",
        required: true,
    },
    realizedSavings: {
        baseName: "realized_savings",
        type: "CommitmentsTimeseriesMetric",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
