// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsUtilizationScalarProductBreakdownEntry.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsUtilizationScalarProductBreakdownEntry = void 0;
/**
 * Per-product utilization data in a scalar utilization response.
 */
class CommitmentsUtilizationScalarProductBreakdownEntry {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsUtilizationScalarProductBreakdownEntry.attributeTypeMap;
    }
}
exports.CommitmentsUtilizationScalarProductBreakdownEntry = CommitmentsUtilizationScalarProductBreakdownEntry;
/**
 * @ignore
 */
CommitmentsUtilizationScalarProductBreakdownEntry.attributeTypeMap = {
    product: {
        baseName: "product",
        type: "string",
        required: true,
    },
    utilization: {
        baseName: "utilization",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
