// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsAzureComputeSPCommitment.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsAzureComputeSPCommitment = void 0;
/**
 * Azure Compute Savings Plan commitment details.
 */
class CommitmentsAzureComputeSPCommitment {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsAzureComputeSPCommitment.attributeTypeMap;
    }
}
exports.CommitmentsAzureComputeSPCommitment = CommitmentsAzureComputeSPCommitment;
/**
 * @ignore
 */
CommitmentsAzureComputeSPCommitment.attributeTypeMap = {
    benefitName: {
        baseName: "benefit_name",
        type: "string",
        required: true,
    },
    commitmentId: {
        baseName: "commitment_id",
        type: "string",
        required: true,
    },
    committedSpendPerHour: {
        baseName: "committed_spend_per_hour",
        type: "number",
        format: "double",
    },
    expirationDate: {
        baseName: "expiration_date",
        type: "string",
    },
    startDate: {
        baseName: "start_date",
        type: "string",
    },
    termLength: {
        baseName: "term_length",
        type: "number",
        format: "double",
    },
    utilization: {
        baseName: "utilization",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
