// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsOnDemandHotspotsScalarResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsOnDemandHotspotsScalarResponse = void 0;
/**
 * Response containing scalar on-demand hot-spots data for cloud commitment programs.
 */
class CommitmentsOnDemandHotspotsScalarResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsOnDemandHotspotsScalarResponse.attributeTypeMap;
    }
}
exports.CommitmentsOnDemandHotspotsScalarResponse = CommitmentsOnDemandHotspotsScalarResponse;
/**
 * @ignore
 */
CommitmentsOnDemandHotspotsScalarResponse.attributeTypeMap = {
    columns: {
        baseName: "columns",
        type: "Array<CommitmentsScalarColumn>",
        required: true,
    },
    meta: {
        baseName: "meta",
        type: "CommitmentsOnDemandHotspotsScalarMeta",
    },
    total: {
        baseName: "total",
        type: "Array<CommitmentsScalarColumn>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
