// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CommitmentsOnDemandHotspotsScalarMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommitmentsOnDemandHotspotsScalarMeta = void 0;
/**
 * Metadata for the on-demand hot-spots scalar response.
 */
class CommitmentsOnDemandHotspotsScalarMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CommitmentsOnDemandHotspotsScalarMeta.attributeTypeMap;
    }
}
exports.CommitmentsOnDemandHotspotsScalarMeta = CommitmentsOnDemandHotspotsScalarMeta;
/**
 * @ignore
 */
CommitmentsOnDemandHotspotsScalarMeta.attributeTypeMap = {
    onDemandFilters: {
        baseName: "on_demand_filters",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
