// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostAnomaliesResponseDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostAnomaliesResponseDataAttributes = void 0;
/**
 * Cost anomaly results and aggregated totals for the queried window.
 */
class CostAnomaliesResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostAnomaliesResponseDataAttributes.attributeTypeMap;
    }
}
exports.CostAnomaliesResponseDataAttributes = CostAnomaliesResponseDataAttributes;
/**
 * @ignore
 */
CostAnomaliesResponseDataAttributes.attributeTypeMap = {
    anomalies: {
        baseName: "anomalies",
        type: "Array<CostAnomaly>",
        required: true,
    },
    avgDailyAnomalousCost: {
        baseName: "avg_daily_anomalous_cost",
        type: "number",
        required: true,
        format: "double",
    },
    totalActualCost: {
        baseName: "total_actual_cost",
        type: "number",
        required: true,
        format: "double",
    },
    totalAnomalousCost: {
        baseName: "total_anomalous_cost",
        type: "number",
        required: true,
        format: "double",
    },
    totalCount: {
        baseName: "total_count",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
