// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagAttributes = void 0;
/**
 * Attributes of a Cloud Cost Management tag.
 */
class CostTagAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagAttributes.attributeTypeMap;
    }
}
exports.CostTagAttributes = CostTagAttributes;
/**
 * @ignore
 */
CostTagAttributes.attributeTypeMap = {
    sources: {
        baseName: "sources",
        type: "Array<string>",
        required: true,
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
