// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostAnomaly.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostAnomaly = void 0;
/**
 * A single detected Cloud Cost Management anomaly.
 */
class CostAnomaly {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostAnomaly.attributeTypeMap;
    }
}
exports.CostAnomaly = CostAnomaly;
/**
 * @ignore
 */
CostAnomaly.attributeTypeMap = {
    actualCost: {
        baseName: "actual_cost",
        type: "number",
        required: true,
        format: "double",
    },
    anomalousCostChange: {
        baseName: "anomalous_cost_change",
        type: "number",
        required: true,
        format: "double",
    },
    anomalyEnd: {
        baseName: "anomaly_end",
        type: "number",
        required: true,
        format: "int64",
    },
    anomalyStart: {
        baseName: "anomaly_start",
        type: "number",
        required: true,
        format: "int64",
    },
    correlatedTags: {
        baseName: "correlated_tags",
        type: "{ [key: string]: Array<string>; }",
        required: true,
    },
    dimensions: {
        baseName: "dimensions",
        type: "{ [key: string]: string; }",
        required: true,
    },
    dismissal: {
        baseName: "dismissal",
        type: "CostAnomalyDismissal",
    },
    maxCost: {
        baseName: "max_cost",
        type: "number",
        required: true,
        format: "double",
    },
    provider: {
        baseName: "provider",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    uuid: {
        baseName: "uuid",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
