// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostAnomalyDismissal.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostAnomalyDismissal = void 0;
/**
 * Resolution metadata for an anomaly that has been dismissed.
 */
class CostAnomalyDismissal {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostAnomalyDismissal.attributeTypeMap;
    }
}
exports.CostAnomalyDismissal = CostAnomalyDismissal;
/**
 * @ignore
 */
CostAnomalyDismissal.attributeTypeMap = {
    cause: {
        baseName: "cause",
        type: "string",
        required: true,
    },
    dismissalId: {
        baseName: "dismissal_id",
        type: "string",
        required: true,
    },
    message: {
        baseName: "message",
        type: "string",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "number",
        required: true,
        format: "int64",
    },
    updatedBy: {
        baseName: "updated_by",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
