// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagKeyDetails.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagKeyDetails = void 0;
/**
 * Additional details for a Cloud Cost Management tag key, including its description and example tag values.
 */
class CostTagKeyDetails {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagKeyDetails.attributeTypeMap;
    }
}
exports.CostTagKeyDetails = CostTagKeyDetails;
/**
 * @ignore
 */
CostTagKeyDetails.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    tagValues: {
        baseName: "tag_values",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
