// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagDescriptionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagDescriptionAttributes = void 0;
/**
 * Human-readable description and metadata attached to a Cloud Cost Management tag key, optionally scoped to a single cloud provider.
 */
class CostTagDescriptionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagDescriptionAttributes.attributeTypeMap;
    }
}
exports.CostTagDescriptionAttributes = CostTagDescriptionAttributes;
/**
 * @ignore
 */
CostTagDescriptionAttributes.attributeTypeMap = {
    cloud: {
        baseName: "cloud",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "CostTagDescriptionSource",
        required: true,
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
