// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateBackfilledDegradationRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBackfilledDegradationRequestDataAttributes = void 0;
/**
 * The supported attributes for creating a backfilled degradation.
 */
class CreateBackfilledDegradationRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateBackfilledDegradationRequestDataAttributes.attributeTypeMap;
    }
}
exports.CreateBackfilledDegradationRequestDataAttributes = CreateBackfilledDegradationRequestDataAttributes;
/**
 * @ignore
 */
CreateBackfilledDegradationRequestDataAttributes.attributeTypeMap = {
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    updates: {
        baseName: "updates",
        type: "Array<CreateBackfilledDegradationRequestDataAttributesUpdatesItems>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
