// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateMaintenanceRequestDataAttributesUpdatesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMaintenanceRequestDataAttributesUpdatesItems = void 0;
/**
 * A maintenance update entry.
 */
class CreateMaintenanceRequestDataAttributesUpdatesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateMaintenanceRequestDataAttributesUpdatesItems.attributeTypeMap;
    }
}
exports.CreateMaintenanceRequestDataAttributesUpdatesItems = CreateMaintenanceRequestDataAttributesUpdatesItems;
/**
 * @ignore
 */
CreateMaintenanceRequestDataAttributesUpdatesItems.attributeTypeMap = {
    componentsAffected: {
        baseName: "components_affected",
        type: "Array<CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems>",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "CreateMaintenanceRequestDataAttributesUpdatesItemsStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
