// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateDegradationRequestDataAttributesUpdatesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDegradationRequestDataAttributesUpdatesItems = void 0;
/**
 * A degradation update entry.
 */
class CreateDegradationRequestDataAttributesUpdatesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateDegradationRequestDataAttributesUpdatesItems.attributeTypeMap;
    }
}
exports.CreateDegradationRequestDataAttributesUpdatesItems = CreateDegradationRequestDataAttributesUpdatesItems;
/**
 * @ignore
 */
CreateDegradationRequestDataAttributesUpdatesItems.attributeTypeMap = {
    componentsAffected: {
        baseName: "components_affected",
        type: "Array<CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems>",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "CreateDegradationRequestDataAttributesStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
