// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems = void 0;
/**
 * A component affected by a degradation update.
 */
class CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems.attributeTypeMap;
    }
}
exports.CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems = CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems;
/**
 * @ignore
 */
CreateDegradationRequestDataAttributesUpdatesItemsComponentsAffectedItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "StatusPagesComponentDataAttributesStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
