// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateBackfilledDegradationRequestDataAttributesUpdatesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBackfilledDegradationRequestDataAttributesUpdatesItems = void 0;
/**
 * A backfilled degradation update entry.
 */
class CreateBackfilledDegradationRequestDataAttributesUpdatesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateBackfilledDegradationRequestDataAttributesUpdatesItems.attributeTypeMap;
    }
}
exports.CreateBackfilledDegradationRequestDataAttributesUpdatesItems = CreateBackfilledDegradationRequestDataAttributesUpdatesItems;
/**
 * @ignore
 */
CreateBackfilledDegradationRequestDataAttributesUpdatesItems.attributeTypeMap = {
    componentsAffected: {
        baseName: "components_affected",
        type: "Array<CreateDegradationRequestDataAttributesComponentsAffectedItems>",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "CreateDegradationRequestDataAttributesStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
