// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateBackfilledMaintenanceRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBackfilledMaintenanceRequestDataAttributes = void 0;
/**
 * The supported attributes for creating a backfilled maintenance.
 */
class CreateBackfilledMaintenanceRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateBackfilledMaintenanceRequestDataAttributes.attributeTypeMap;
    }
}
exports.CreateBackfilledMaintenanceRequestDataAttributes = CreateBackfilledMaintenanceRequestDataAttributes;
/**
 * @ignore
 */
CreateBackfilledMaintenanceRequestDataAttributes.attributeTypeMap = {
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    updates: {
        baseName: "updates",
        type: "[CreateBackfilledMaintenanceRequestDataAttributesUpdatesItems, CreateBackfilledMaintenanceRequestDataAttributesUpdatesItems]",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
