// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreatePublishRequestRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePublishRequestRequestDataAttributes = void 0;
/**
 * Attributes for creating a publish request.
 */
class CreatePublishRequestRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreatePublishRequestRequestDataAttributes.attributeTypeMap;
    }
}
exports.CreatePublishRequestRequestDataAttributes = CreatePublishRequestRequestDataAttributes;
/**
 * @ignore
 */
CreatePublishRequestRequestDataAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
