// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems = void 0;
/**
 * A component affected by a maintenance update.
 */
class CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems.attributeTypeMap;
    }
}
exports.CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems = CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems;
/**
 * @ignore
 */
CreateMaintenanceRequestDataAttributesUpdatesItemsComponentsAffectedItems.attributeTypeMap = {
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "PatchMaintenanceRequestDataAttributesComponentsAffectedItemsStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
