// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FeatureFlagEnvironmentListItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureFlagEnvironmentListItem = void 0;
/**
 * Environment-specific settings for a feature flag in list responses.
 */
class FeatureFlagEnvironmentListItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FeatureFlagEnvironmentListItem.attributeTypeMap;
    }
}
exports.FeatureFlagEnvironmentListItem = FeatureFlagEnvironmentListItem;
/**
 * @ignore
 */
FeatureFlagEnvironmentListItem.attributeTypeMap = {
    defaultAllocationKey: {
        baseName: "default_allocation_key",
        type: "string",
    },
    defaultVariantId: {
        baseName: "default_variant_id",
        type: "string",
    },
    environmentId: {
        baseName: "environment_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    environmentName: {
        baseName: "environment_name",
        type: "string",
    },
    environmentQueries: {
        baseName: "environment_queries",
        type: "Array<string>",
    },
    isProduction: {
        baseName: "is_production",
        type: "boolean",
    },
    overrideAllocationKey: {
        baseName: "override_allocation_key",
        type: "string",
    },
    overrideVariantId: {
        baseName: "override_variant_id",
        type: "string",
    },
    pendingSuggestionId: {
        baseName: "pending_suggestion_id",
        type: "string",
    },
    requireFeatureFlagApproval: {
        baseName: "require_feature_flag_approval",
        type: "boolean",
    },
    status: {
        baseName: "status",
        type: "FeatureFlagStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
