// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IssueRegression.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IssueRegression = void 0;
/**
 * Regression information for an issue that was previously resolved and then reopened.
 */
class IssueRegression {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IssueRegression.attributeTypeMap;
    }
}
exports.IssueRegression = IssueRegression;
/**
 * @ignore
 */
IssueRegression.attributeTypeMap = {
    regressedAt: {
        baseName: "regressed_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    regressedAtVersion: {
        baseName: "regressed_at_version",
        type: "string",
    },
    resolvedAt: {
        baseName: "resolved_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
