// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueDataAttributesRequest = void 0;
/**
 * Attributes for creating an LLM Observability annotation queue.
 */
class LLMObsAnnotationQueueDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueDataAttributesRequest = LLMObsAnnotationQueueDataAttributesRequest;
/**
 * @ignore
 */
LLMObsAnnotationQueueDataAttributesRequest.attributeTypeMap = {
    annotationSchema: {
        baseName: "annotation_schema",
        type: "LLMObsAnnotationSchema",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    projectId: {
        baseName: "project_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
