// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueLabelSchemaData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueLabelSchemaData = void 0;
/**
 * Data object for an annotation queue label schema.
 */
class LLMObsAnnotationQueueLabelSchemaData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueLabelSchemaData.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueLabelSchemaData = LLMObsAnnotationQueueLabelSchemaData;
/**
 * @ignore
 */
LLMObsAnnotationQueueLabelSchemaData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsAnnotationQueueLabelSchemaAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotationQueueType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
