// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsLabelSchema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsLabelSchema = void 0;
/**
 * Schema definition for a single label in an annotation queue.
 */
class LLMObsLabelSchema {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsLabelSchema.attributeTypeMap;
    }
}
exports.LLMObsLabelSchema = LLMObsLabelSchema;
/**
 * @ignore
 */
LLMObsLabelSchema.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    hasAssessment: {
        baseName: "has_assessment",
        type: "boolean",
    },
    hasReasoning: {
        baseName: "has_reasoning",
        type: "boolean",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    isAssessment: {
        baseName: "is_assessment",
        type: "boolean",
    },
    isInteger: {
        baseName: "is_integer",
        type: "boolean",
    },
    isRequired: {
        baseName: "is_required",
        type: "boolean",
    },
    max: {
        baseName: "max",
        type: "number",
        format: "double",
    },
    min: {
        baseName: "min",
        type: "number",
        format: "double",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsLabelSchemaType",
        required: true,
    },
    values: {
        baseName: "values",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
