// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationQueueUpdateDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationQueueUpdateDataAttributesRequest = void 0;
/**
 * Attributes for updating an LLM Observability annotation queue. All fields are optional.
 */
class LLMObsAnnotationQueueUpdateDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationQueueUpdateDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsAnnotationQueueUpdateDataAttributesRequest = LLMObsAnnotationQueueUpdateDataAttributesRequest;
/**
 * @ignore
 */
LLMObsAnnotationQueueUpdateDataAttributesRequest.attributeTypeMap = {
    annotationSchema: {
        baseName: "annotation_schema",
        type: "LLMObsAnnotationSchema",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
