// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MuteFindingsMuteAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuteFindingsMuteAttributes = void 0;
/**
 * Mute properties to apply to the findings.
 */
class MuteFindingsMuteAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MuteFindingsMuteAttributes.attributeTypeMap;
    }
}
exports.MuteFindingsMuteAttributes = MuteFindingsMuteAttributes;
/**
 * @ignore
 */
MuteFindingsMuteAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    expireAt: {
        baseName: "expire_at",
        type: "number",
        format: "int64",
    },
    isMuted: {
        baseName: "is_muted",
        type: "boolean",
        required: true,
    },
    reason: {
        baseName: "reason",
        type: "MuteFindingsReason",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
