// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MuteFindingsRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuteFindingsRequestData = void 0;
/**
 * Data of the mute request.
 */
class MuteFindingsRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MuteFindingsRequestData.attributeTypeMap;
    }
}
exports.MuteFindingsRequestData = MuteFindingsRequestData;
/**
 * @ignore
 */
MuteFindingsRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "MuteFindingsRequestDataAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
    },
    relationships: {
        baseName: "relationships",
        type: "MuteFindingsRequestDataRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "MuteDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
