// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgAttributes = void 0;
/**
 * Attributes of an organization.
 */
class OrgAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgAttributes.attributeTypeMap;
    }
}
exports.OrgAttributes = OrgAttributes;
/**
 * @ignore
 */
OrgAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    disabled: {
        baseName: "disabled",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    publicId: {
        baseName: "public_id",
        type: "string",
        required: true,
    },
    sharing: {
        baseName: "sharing",
        type: "string",
        required: true,
    },
    url: {
        baseName: "url",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
