// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RuleBasedViewComplianceFramework.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleBasedViewComplianceFramework = void 0;
/**
 * Compliance framework mapping for a rule.
 */
class RuleBasedViewComplianceFramework {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RuleBasedViewComplianceFramework.attributeTypeMap;
    }
}
exports.RuleBasedViewComplianceFramework = RuleBasedViewComplianceFramework;
/**
 * @ignore
 */
RuleBasedViewComplianceFramework.attributeTypeMap = {
    control: {
        baseName: "control",
        type: "string",
    },
    framework: {
        baseName: "framework",
        type: "string",
    },
    isDefault: {
        baseName: "is_default",
        type: "boolean",
    },
    message: {
        baseName: "message",
        type: "string",
    },
    requirement: {
        baseName: "requirement",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
