// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RuleBasedViewRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleBasedViewRule = void 0;
/**
 * A compliance rule along with its evaluation statistics and framework mappings.
 */
class RuleBasedViewRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RuleBasedViewRule.attributeTypeMap;
    }
}
exports.RuleBasedViewRule = RuleBasedViewRule;
/**
 * @ignore
 */
RuleBasedViewRule.attributeTypeMap = {
    complianceFrameworks: {
        baseName: "compliance_frameworks",
        type: "Array<RuleBasedViewComplianceFramework>",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    resourceAttributes: {
        baseName: "resourceAttributes",
        type: "Array<string>",
        required: true,
    },
    resourceCategory: {
        baseName: "resourceCategory",
        type: "string",
        required: true,
    },
    resourceType: {
        baseName: "resourceType",
        type: "string",
        required: true,
    },
    stats: {
        baseName: "stats",
        type: "RuleBasedViewRuleStats",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "RuleBasedViewRuleCategory",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
