// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RuleBasedViewRuleStats.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleBasedViewRuleStats = void 0;
/**
 * Counts of findings for the rule, grouped by their evaluation status.
 */
class RuleBasedViewRuleStats {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RuleBasedViewRuleStats.attributeTypeMap;
    }
}
exports.RuleBasedViewRuleStats = RuleBasedViewRuleStats;
/**
 * @ignore
 */
RuleBasedViewRuleStats.attributeTypeMap = {
    fail: {
        baseName: "fail",
        type: "number",
        required: true,
        format: "int64",
    },
    muted: {
        baseName: "muted",
        type: "number",
        required: true,
        format: "int64",
    },
    pass: {
        baseName: "pass",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
