// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RulesetStatusRespDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RulesetStatusRespDataAttributes = void 0;
/**
 * Processing status for a tag pipeline ruleset.
 */
class RulesetStatusRespDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RulesetStatusRespDataAttributes.attributeTypeMap;
    }
}
exports.RulesetStatusRespDataAttributes = RulesetStatusRespDataAttributes;
/**
 * @ignore
 */
RulesetStatusRespDataAttributes.attributeTypeMap = {
    processingStatus: {
        baseName: "processing_status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
