// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringRuleBulkDeleteResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringRuleBulkDeleteResponse = void 0;
/**
 * Response for bulk deleting security monitoring rules.
 */
class SecurityMonitoringRuleBulkDeleteResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringRuleBulkDeleteResponse.attributeTypeMap;
    }
}
exports.SecurityMonitoringRuleBulkDeleteResponse = SecurityMonitoringRuleBulkDeleteResponse;
/**
 * @ignore
 */
SecurityMonitoringRuleBulkDeleteResponse.attributeTypeMap = {
    deletedRules: {
        baseName: "deletedRules",
        type: "Array<string>",
    },
    failedRules: {
        baseName: "failedRules",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
