// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeTimeSlotRecurrenceResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeTimeSlotRecurrenceResponse = void 0;
/**
 * Recurrence settings returned in a Synthetics downtime time slot response.
 */
class SyntheticsDowntimeTimeSlotRecurrenceResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeTimeSlotRecurrenceResponse.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeTimeSlotRecurrenceResponse = SyntheticsDowntimeTimeSlotRecurrenceResponse;
/**
 * @ignore
 */
SyntheticsDowntimeTimeSlotRecurrenceResponse.attributeTypeMap = {
    frequency: {
        baseName: "frequency",
        type: "SyntheticsDowntimeFrequency",
        required: true,
    },
    interval: {
        baseName: "interval",
        type: "number",
        required: true,
        format: "int64",
    },
    until: {
        baseName: "until",
        type: "SyntheticsDowntimeTimeSlotDate",
    },
    weekdays: {
        baseName: "weekdays",
        type: "Array<SyntheticsDowntimeWeekday>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
