// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeDataAttributesRequest = void 0;
/**
 * Attributes for creating or updating a Synthetics downtime.
 */
class SyntheticsDowntimeDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeDataAttributesRequest.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeDataAttributesRequest = SyntheticsDowntimeDataAttributesRequest;
/**
 * @ignore
 */
SyntheticsDowntimeDataAttributesRequest.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    isEnabled: {
        baseName: "isEnabled",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    testIds: {
        baseName: "testIds",
        type: "Array<string>",
        required: true,
    },
    timeSlots: {
        baseName: "timeSlots",
        type: "Array<SyntheticsDowntimeTimeSlotRequest>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
