// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeDataAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeDataAttributesResponse = void 0;
/**
 * Attributes of a Synthetics downtime response object.
 */
class SyntheticsDowntimeDataAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeDataAttributesResponse.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeDataAttributesResponse = SyntheticsDowntimeDataAttributesResponse;
/**
 * @ignore
 */
SyntheticsDowntimeDataAttributesResponse.attributeTypeMap = {
    createdAt: {
        baseName: "createdAt",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "createdBy",
        type: "string",
        required: true,
    },
    createdByName: {
        baseName: "createdByName",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    isEnabled: {
        baseName: "isEnabled",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
        required: true,
    },
    testIds: {
        baseName: "testIds",
        type: "Array<string>",
        required: true,
    },
    timeSlots: {
        baseName: "timeSlots",
        type: "Array<SyntheticsDowntimeTimeSlotResponse>",
        required: true,
    },
    updatedAt: {
        baseName: "updatedAt",
        type: "Date",
        required: true,
        format: "date-time",
    },
    updatedBy: {
        baseName: "updatedBy",
        type: "string",
        required: true,
    },
    updatedByName: {
        baseName: "updatedByName",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
