// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeTimeSlotResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeTimeSlotResponse = void 0;
/**
 * A time slot returned in a Synthetics downtime response.
 */
class SyntheticsDowntimeTimeSlotResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeTimeSlotResponse.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeTimeSlotResponse = SyntheticsDowntimeTimeSlotResponse;
/**
 * @ignore
 */
SyntheticsDowntimeTimeSlotResponse.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    recurrence: {
        baseName: "recurrence",
        type: "SyntheticsDowntimeTimeSlotRecurrenceResponse",
    },
    start: {
        baseName: "start",
        type: "SyntheticsDowntimeTimeSlotDate",
        required: true,
    },
    timezone: {
        baseName: "timezone",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
