// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeTimeSlotDate.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeTimeSlotDate = void 0;
/**
 * A specific date and time used to define the start or end of a Synthetics downtime time slot.
 */
class SyntheticsDowntimeTimeSlotDate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeTimeSlotDate.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeTimeSlotDate = SyntheticsDowntimeTimeSlotDate;
/**
 * @ignore
 */
SyntheticsDowntimeTimeSlotDate.attributeTypeMap = {
    day: {
        baseName: "day",
        type: "number",
        required: true,
        format: "int64",
    },
    hour: {
        baseName: "hour",
        type: "number",
        required: true,
        format: "int64",
    },
    minute: {
        baseName: "minute",
        type: "number",
        required: true,
        format: "int64",
    },
    month: {
        baseName: "month",
        type: "number",
        required: true,
        format: "int64",
    },
    year: {
        baseName: "year",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
