// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeTimeSlotRecurrenceRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeTimeSlotRecurrenceRequest = void 0;
/**
 * Recurrence settings for a Synthetics downtime time slot.
 */
class SyntheticsDowntimeTimeSlotRecurrenceRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeTimeSlotRecurrenceRequest.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeTimeSlotRecurrenceRequest = SyntheticsDowntimeTimeSlotRecurrenceRequest;
/**
 * @ignore
 */
SyntheticsDowntimeTimeSlotRecurrenceRequest.attributeTypeMap = {
    end: {
        baseName: "end",
        type: "SyntheticsDowntimeTimeSlotDate",
    },
    frequency: {
        baseName: "frequency",
        type: "SyntheticsDowntimeFrequency",
        required: true,
    },
    interval: {
        baseName: "interval",
        type: "number",
        format: "int64",
    },
    weekdays: {
        baseName: "weekdays",
        type: "Array<SyntheticsDowntimeWeekday>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
