// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SyntheticsDowntimeTimeSlotRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsDowntimeTimeSlotRequest = void 0;
/**
 * A time slot for a Synthetics downtime create or update request.
 */
class SyntheticsDowntimeTimeSlotRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsDowntimeTimeSlotRequest.attributeTypeMap;
    }
}
exports.SyntheticsDowntimeTimeSlotRequest = SyntheticsDowntimeTimeSlotRequest;
/**
 * @ignore
 */
SyntheticsDowntimeTimeSlotRequest.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "number",
        required: true,
        format: "int64",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    recurrence: {
        baseName: "recurrence",
        type: "SyntheticsDowntimeTimeSlotRecurrenceRequest",
    },
    start: {
        baseName: "start",
        type: "SyntheticsDowntimeTimeSlotDate",
        required: true,
    },
    timezone: {
        baseName: "timezone",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
