// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpdateAppSelfServiceRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAppSelfServiceRequestDataAttributes = void 0;
/**
 * Attributes for updating an app's self-service status.
 */
class UpdateAppSelfServiceRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpdateAppSelfServiceRequestDataAttributes.attributeTypeMap;
    }
}
exports.UpdateAppSelfServiceRequestDataAttributes = UpdateAppSelfServiceRequestDataAttributes;
/**
 * @ignore
 */
UpdateAppSelfServiceRequestDataAttributes.attributeTypeMap = {
    selfService: {
        baseName: "selfService",
        type: "boolean",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
