// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpdateAppTagsRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAppTagsRequestDataAttributes = void 0;
/**
 * Attributes for replacing an app's tags.
 */
class UpdateAppTagsRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpdateAppTagsRequestDataAttributes.attributeTypeMap;
    }
}
exports.UpdateAppTagsRequestDataAttributes = UpdateAppTagsRequestDataAttributes;
/**
 * @ignore
 */
UpdateAppTagsRequestDataAttributes.attributeTypeMap = {
    tags: {
        baseName: "tags",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
