// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpdateAppFavoriteRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAppFavoriteRequestDataAttributes = void 0;
/**
 * Attributes for updating an app's favorite status.
 */
class UpdateAppFavoriteRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpdateAppFavoriteRequestDataAttributes.attributeTypeMap;
    }
}
exports.UpdateAppFavoriteRequestDataAttributes = UpdateAppFavoriteRequestDataAttributes;
/**
 * @ignore
 */
UpdateAppFavoriteRequestDataAttributes.attributeTypeMap = {
    favorite: {
        baseName: "favorite",
        type: "boolean",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
