// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpsertCloudInventorySyncConfigRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertCloudInventorySyncConfigRequestAttributes = void 0;
/**
 * Settings for the cloud provider specified in `data.id`. Include only the matching provider object (`aws`, `gcp`, or `azure`).
 */
class UpsertCloudInventorySyncConfigRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpsertCloudInventorySyncConfigRequestAttributes.attributeTypeMap;
    }
}
exports.UpsertCloudInventorySyncConfigRequestAttributes = UpsertCloudInventorySyncConfigRequestAttributes;
/**
 * @ignore
 */
UpsertCloudInventorySyncConfigRequestAttributes.attributeTypeMap = {
    aws: {
        baseName: "aws",
        type: "CloudInventorySyncConfigAWSRequestAttributes",
    },
    azure: {
        baseName: "azure",
        type: "CloudInventorySyncConfigAzureRequestAttributes",
    },
    gcp: {
        baseName: "gcp",
        type: "CloudInventorySyncConfigGCPRequestAttributes",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
