// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpsertCloudInventorySyncConfigRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertCloudInventorySyncConfigRequestData = void 0;
/**
 * Storage Management configuration data for the create or update request.
 */
class UpsertCloudInventorySyncConfigRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpsertCloudInventorySyncConfigRequestData.attributeTypeMap;
    }
}
exports.UpsertCloudInventorySyncConfigRequestData = UpsertCloudInventorySyncConfigRequestData;
/**
 * @ignore
 */
UpsertCloudInventorySyncConfigRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "UpsertCloudInventorySyncConfigRequestAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "CloudInventoryCloudProviderId",
        required: true,
    },
    type: {
        baseName: "type",
        type: "CloudInventoryCloudProviderRequestType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
