// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ValidateV2Attributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidateV2Attributes = void 0;
/**
 * Attributes of the API key validation response.
 */
class ValidateV2Attributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ValidateV2Attributes.attributeTypeMap;
    }
}
exports.ValidateV2Attributes = ValidateV2Attributes;
/**
 * @ignore
 */
ValidateV2Attributes.attributeTypeMap = {
    apiKeyId: {
        baseName: "api_key_id",
        type: "string",
        required: true,
    },
    apiKeyScopes: {
        baseName: "api_key_scopes",
        type: "Array<string>",
        required: true,
    },
    valid: {
        baseName: "valid",
        type: "boolean",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
