// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebIntegrationAccountResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebIntegrationAccountResponseData = void 0;
/**
 * Data object of a web integration account.
 */
class WebIntegrationAccountResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebIntegrationAccountResponseData.attributeTypeMap;
    }
}
exports.WebIntegrationAccountResponseData = WebIntegrationAccountResponseData;
/**
 * @ignore
 */
WebIntegrationAccountResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "WebIntegrationAccountResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "WebIntegrationAccountType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
