// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebIntegrationAccountUpdateRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebIntegrationAccountUpdateRequestAttributes = void 0;
/**
 * Attributes object for updating a web integration account.
 */
class WebIntegrationAccountUpdateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebIntegrationAccountUpdateRequestAttributes.attributeTypeMap;
    }
}
exports.WebIntegrationAccountUpdateRequestAttributes = WebIntegrationAccountUpdateRequestAttributes;
/**
 * @ignore
 */
WebIntegrationAccountUpdateRequestAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    secrets: {
        baseName: "secrets",
        type: "{ [key: string]: any; }",
    },
    settings: {
        baseName: "settings",
        type: "{ [key: string]: any; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
