// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebIntegrationAccountCreateRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebIntegrationAccountCreateRequestAttributes = void 0;
/**
 * Attributes object for creating a web integration account.
 */
class WebIntegrationAccountCreateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebIntegrationAccountCreateRequestAttributes.attributeTypeMap;
    }
}
exports.WebIntegrationAccountCreateRequestAttributes = WebIntegrationAccountCreateRequestAttributes;
/**
 * @ignore
 */
WebIntegrationAccountCreateRequestAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    secrets: {
        baseName: "secrets",
        type: "{ [key: string]: any; }",
        required: true,
    },
    settings: {
        baseName: "settings",
        type: "{ [key: string]: any; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
