// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebIntegrationAccountResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebIntegrationAccountResponseAttributes = void 0;
/**
 * Attributes object of a web integration account. Secrets are never returned.
 */
class WebIntegrationAccountResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebIntegrationAccountResponseAttributes.attributeTypeMap;
    }
}
exports.WebIntegrationAccountResponseAttributes = WebIntegrationAccountResponseAttributes;
/**
 * @ignore
 */
WebIntegrationAccountResponseAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    settings: {
        baseName: "settings",
        type: "{ [key: string]: any; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
